hangman/game.py

```diff
@@ -1,13 +1,101 @@
 from .exceptions import *
+import random
 
 
 class GuessAttempt(object):
-    pass
-
+    new_word = ""
+    def __init__(self, guess, hit=False, miss=False):
+        self.hit = hit
+        self.miss = miss
+        self.guess = guess
+        if self.hit == True and self.miss== True:
+            raise InvalidGuessAttempt
+        
+            
+    def is_hit(self):
+        return self.hit
+        
+    def is_miss(self):
+        return self.miss
 
 class GuessWord(object):
-    pass
+    def __init__(self, answer):
+        self.answer = answer.lower()
+        self.masked = '*' * len(self.answer)
+        if not answer:
+            raise InvalidWordException
+            
+   
+     
+    def perform_attempt(self, guess):
+        self.guess = guess.lower()
+        if len(self.guess) > 1:
+            raise InvalidGuessedLetterException
+         
+        if self.guess not in self.answer:
+            return GuessAttempt(self.guess, miss=True)
+        
+        new_word = ""
+    
+        for i in range(len(self.answer)):
+            if self.answer[i] == self.guess:
```

Comment:
You're killing me with this for loop man 😥. Remember you can do `for i, answer_char in enumerate(self.answer)`.

Even better, check the `zip` function: https://repl.it/repls/SparseVastStarling